usaspending_api/common/tests/unit/test_tracing.py

```diff
@@ -0,0 +1,207 @@
+import inspect
+
+import ddtrace
+import logging
+import multiprocessing as mp
+import pytest
+
+from logging.handlers import QueueHandler
+from _pytest.logging import LogCaptureFixture
+from ddtrace.ext import SpanTypes
+
+from usaspending_api.common.tracing import DatadogEagerlyDropTraceFilter, DatadogLoggingTraceFilter, SubprocessTrace
+
+
+@pytest.fixture
+def datadog_tracer() -> ddtrace.Tracer:
+    """Fixture to temporarily enable the Datadog Tracer during a test"""
+    tracer_state = ddtrace.tracer.enabled
+    ddtrace.tracer.enabled = True
+    yield ddtrace.tracer
+    ddtrace.tracer.enabled = tracer_state
+
+
+@pytest.yield_fixture
+def caplog(caplog):
+    """A decorator (pattern) fixture around the pytest caplog fixture that adds the ability to temporarily alter
+    loggers with propagate=False to True for duration of the test, so their output is propagated to the caplog log
+    handler"""
+
+    restore = []
+    for logger in logging.Logger.manager.loggerDict.values():
+        try:
+            if not logger.propagate:
+                logger.propagate = True
+                restore += [logger]
+        except AttributeError:
+            pass
+    yield caplog
+    for logger in restore:
+        logger.propagate = False
+
+
+def test_logging_trace_spans(datadog_tracer: ddtrace.Tracer, caplog: LogCaptureFixture):
```

Comment:
Do type hints help for pytests?

Reply:
So that I could see autocomplete in the IDE for the `caplog` fixture